### PR TITLE
fix/dataops-523

### PR DESCRIPTION
**Pull request by jordan-springer**
this one took me way too long to crack, essentially: the `user_type: not_null` was causing a failure bc we needed to fully rebuild our user-based models to remove the `null` rows. then, then we

### CODE REVIEW

File: dbt/models/marts/metrics/_metrics__models.yml [hunk 1]
```diff
@@ -15,38 +14,50 @@ models:
...
-        tests:
-          - not_null
+        # tests:
+        #   - not_null
+      
       - name: num_schools_this_week
         description: total number of schools within the school level and active status dimensions within that week
         tests:
-          - not_null
+          # - not_null
```
Comment (jordan-springer) [inline dbt/models/marts/metrics/_metrics__models.yml:48]: this is an example of using tests to anticipate variance

File: dbt/models/marts/metrics/_metrics__models.yml [hunk 2]
```diff
@@ -60,7 +71,13 @@ models:
       - name: user_type
         description: teacher or student
         tests:
-          - not_null
+          - unique:
```
Comment (jordan-springer) [inline dbt/models/marts/metrics/_metrics__models.yml:74]: this is (essentially) the patch: `user_type` is not "unique," is one of two accepted values (see comments).
so what we added here is a warning if there are 3 types (i.e. a `null` got through somehow), and if there are more than that, throw an error (exception.)

File: dbt/models/marts/users/_users__models.yml [hunk 3]
```diff
@@ -9,7 +9,13 @@ models:
         description: UID for each user
         tests: 
           - not_null
-          
+
+      - name: user_type
+        description: student or teacherm if NULL = fail
+            # - accepted_values:
```
Comment (jordan-springer) [inline dbt/models/marts/users/_users__models.yml:15]: example here if we want to pursue this route

File: dbt/models/marts/metrics/_metrics__models.yml [hunk 4]
```diff
@@ -97,8 +114,12 @@ models:
     columns:
       - name: user_type
         description: teacher or student
-        tests: 
-          - not_null
+        tests:
+          - unique:
```
Comment (jordan-springer) [inline dbt/models/marts/metrics/_metrics__models.yml:118]: oops, need to remove this

File: dbt/models/marts/metrics/_metrics__models.yml [hunk 5]
```diff
@@ -97,8 +114,12 @@ models:
...
+              config:
+                severity: error
+                error_if: ">10"
```
Comment (jordan-springer) [inline dbt/models/marts/metrics/_metrics__models.yml:121]:
```suggestion
                error_if: ">3"
```

File: dbt/models/marts/metrics/_metrics__models.yml [hunk 6]
```diff
@@ -15,38 +14,50 @@ models:
...
       - name: num_schools_this_week
         description: total number of schools within the school level and active status dimensions within that week
         tests:
-          - not_null
+          # - not_null
+          - unique:
+              config:
+                severity: warn
+                # error_if: "<1"
+                warn_if: "<10"
```
Comment (jordan-springer) [inline dbt/models/marts/metrics/_metrics__models.yml:53]: proof of concept for analytics testing

File: dbt/models/marts/metrics/_metrics__models.yml [hunk 7]
```diff
@@ -97,8 +114,12 @@ models:
...
       - name: user_type
         description: teacher or student
-        tests: 
-          - not_null
+        tests:
+          - unique:
+              config:
+                severity: error
+                error_if: ">10"
+                warn_if: "<10"
```
Comment (jordan-springer) [inline dbt/models/marts/metrics/_metrics__models.yml:122]:
```suggestion
                warn_if: "=3"
```

**Review by jordan-springer — COMMENTED** [review]
let me know if you have any questions!

**Comment by jordan-springer** [comment]
build ran successfully, will merge in the morning

**Comment by jordan-springer** [comment]
see jira for summary points

File: dbt/models/marts/users/_users__models.yml (hunk 3 above)
Comment (nataliazm99) [reply dbt/models/marts/users/_users__models.yml:15]: There's a typo in teacher (m at the end)
I suggest being more explicit: description: allowed values: student or teacher;  if NULL = fail

**Pull request merged**